feat(stats-detectors): Add initial docs for function regression issues (#8402)
getsentry/sentry-docs: src/docs/product/issues/issue-details/performance-issues/regressions.mdx

> ---

Reviewer: unless we want both statistical detector issues on the same page, could you change the route from `regressions` to `function-regressions`?

title: "Function Regression"
sidebar_order: 40

> description: "Learn more about Function Regression Issue and how to diagnose and fix them."

Reviewer 2: ```suggestion
description: "Learn more about Function Regression issues and how to diagnose and fix them."
```

---

> 

Reviewer 3: I believe we should include the EA note while this is in EA.

```suggestion

<Include name="feature-available-for-user-group-early-adopter.mdx" />

```

Reviewer 3: And can we add a link to this page to the list on the Early Adopter Features page: src/docs/product/accounts/early-adopter-features/index.mdx ?

Function Regression issues capture a generic class of problems where the duration of a function has increased over time. This can lead to degraded performance in your application. Sentry attempts to monitor some of your most common functions out of the box and report any possible regressions under the Function Regression issue type.

## Detection Criteria

> [Profiling](/product/profiling) must be enabled for Sentry to detect Function Regression issues. Once set up, Sentry will periodically check the 95th percentile function duration of the most common application functions in your project. When a significant change in the function duration for a function is detected and it has sustained for some time, a regression issue is reported.

Reviewer 2: ```suggestion
[Profiling](/product/profiling) must be enabled for Sentry to detect Function Regression issues. Once set up, Sentry will periodically check the 95th percentile function duration of the most common application functions in your project. When a significant change in the p95 value for function duration is detected and has been sustained for some time, a regression issue is reported.
```

> [Profiling](/product/profiling) must be enabled for Sentry to detect Function Regression issues. Once set up, Sentry will periodically check the 95th percentile function duration of the most common application functions in your project. When a significant change in the p95 value for function duration is detected and has been sustained for some time, a regression issue is reported.

Reviewer 3: ```suggestion
Sentry can only detect Function Regression issues on projects that have [Profiling](/product/profiling) enabled. Sentry will periodically check the 95th percentile function duration of the most common application functions in your project. When a significant increase in the p95 value for function duration is detected and has been sustained for some time, a regression issue is created.
```

Can we specify what "some time" is?

> ## Charts

Reviewer 3: Missing screenshots?

Author: Will be added for GA as we're still polishing the UI

> The chart shows a window of up to 14 days before and after where the regression was identified. Here you can see the p95 function duration as well as the function throughput.

Reviewer 2: ```suggestion
The chart shows a window of up to 14 days before and after the regression was identified. Here, you can see the p95 function duration as well as the function throughput.
```

> A common cause of regressions is an increase in throughput of a function in the code path that contains a mutex which is now causing lock contention.

Reviewer 2: ```suggestion
A common cause of regressions is an increase in the throughput of a function. For example, when the throughput increases in a code path that contains a mutex, this can cause lock contention.
```
found the original kinda hard to read with all the of, in, that, which... hope this suggestion is still accurate

> A common cause of regressions is an increase in the throughput of a function. For example, when the throughput increases in a code path that contains a mutex, this can cause lock contention.

Reviewer 3: ```suggestion
A common cause of regressions is an increase in the throughput of a function. For example, when throughput increases in a code path that contains a mutex, this can cause lock contention.
```

## Function Evidence

To find additional information about your regressed function, go to the "Function Evidence" section which shows the following:

- **Function Name:** The name of the function that has regressed.
- **Package Name:** The package of the function that has regressed.
- **File Name:** The file name of the function that has regressed.
- **Regression Date:** The date and time at which the regression started.

## Transactions Impacted

> The "Transactions Impacted" section shows a list of transactions where this regression may be observed. Next to each transaction, you can find the following:

Reviewer 3: ```suggestion
The "Transactions Impacted" section shows a list of transactions where this regression shows up. Next to each transaction, you can find the following:
```

> - A sparkline showing the 95th percentile function duration of the regressed function within the transaction.

Reviewer 2: ```suggestion
- A sparkline showing the p95 function duration of the regressed function within the transaction.
```

I feel like p95 is pretty common and doesn't need to be spelled out after the first mention

> - The 95th percentile function duration of the regressed function before and after the regression.

Reviewer 2: ```suggestion
- The p95 function duration of the regressed function before and after the regression.
```

> The before and after values also link to example profiles that you can examine to identify the root cause.

Reviewer 3: ```suggestion
The before and after values also link to example profiles you can examine to identify the root cause of the regression.
```
